### ISSUE TITLE
Add button for copying code

### ISSUE THREAD

**Issue by kaycebasques** [GoogleChrome/web.dev#1871]
**Is your feature request related to a problem? Please describe.**

Blocks of code like this should be copy-able with a single button click:

![image](https://user-images.githubusercontent.com/4713486/68536500-c3ee0880-0308-11ea-9b02-80f6b5457b83.png)

**Describe the solution you'd like**

Add a **Copy to clipboard** button to the top-right of each code block.

**Describe alternatives you've considered**

None.

**Additional context**

See DevSite for example:

![image](https://user-images.githubusercontent.com/4713486/68536505-eed85c80-0308-11ea-93a5-efb548c8a963.png)

https://cloud.google.com/nodejs/getting-started/hello-world

Would also be helpful to fire an analytics event every time that the code is copied. This would enable us to track how many users actually copy the code when doing a codelab, for example.

**Comment by robdodson** [comment]
I think we can add this if folks want it, though it was a conscious decision to not include it in the first place.

We originally wanted to encourage folks to type the code when they're doing codelabs, based on an idea I stole from Learn Python the Hard Way that if you're copy/pasting in a tutorial then you aren't really engaging your brain.

**Comment by MichaelSolati** [comment]
I personally feel like it would be useful. While I agree that having people type in the code can reinforce the learning process, users can still just select the text, then copy and paste. So it adds a small barrier that's easily overcome.

Also looking over some of the code, it seems as if we are showing changes to the code with the context of where it is changed. If they were to have an easier copy and paste experience, they'll still need to make sure they address duplicate/redundant code (which may have them looking at the code snippet a bit longer to understand what's going on).

Tl:dr => The barrier to copy and paste is minimal, so is it needed? Also because of redundant code we likely are keeping users engaged.

**Comment by robdodson** [comment]
yep, sounds like we should add it
